File: server/src/models/EFPLStats.js
const mongoose = require("mongoose");

const dreamTeamSchema = mongoose.Schema({
  goalKeeper: [],
  defenders: [],
  midfielders: [],
  attackers: [],
});

const efplGameWeekSchema = mongoose.Schema({
  gameWeekNumber: {
    type: Number,
  },
  highestPoint: {
    type: Number,
  },
  averagePoint: {
    type: Number,
  },
  mostSelectedPlayer: {
    type: Array,
    default: [],
  },
  mostTransferredOutPlayer: {
    type: Array,
    default: [],
  },
  transfersMadeCount: {
    type: Number,
    default: 0,
  },
  mostCaptainedPlayer: { type: Array, default: [] },
  mostViceCaptainedPlayer: {
    type: Array,
    default: [],
  },
  benchBoostCount: {
    type: Number,
    default: 0,
  },
  freeHitCount: {
    type: Number,
    default: 0,
  },
  wildCardCount: {
    type: Number,
    default: 0,
  },
  tripleCaptainCount: {
    type: Number,
    default: 0,
  },
  dreamTeam: {
    type: dreamTeamSchema,
  },
});

const efplStatsSchema = mongoose.Schema({
  allStats: efplGameWeekSchema,
  gameWeekNumber: { type: Number },
});

module.exports = mongoose.model("EFPLStat", efplStatsSchema);

File: flutter-user-client/lib/domain/efpl_stats/efpl_stats.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'efpl_stats.freezed.dart';

@freezed
abstract class EFPLStats with _$EFPLStats {
  const factory EFPLStats({
    required int highestPoint,
    required int averagePoint,
    required String mostSelectedPlayer,
    required String mostTransferredOutPlayer,
    required int transfersMadeCount,
    required String mostCaptainedPlayer,
    required String mostViceCaptainedPlayer,
    required int benchBoostCount,
    required int freeHitCount,
    required int wildCardCount,
    required int tripleCaptainCount,
    required Map dreamTeam,
    required int maxActiveCount,
    required int gameWeekId,
  }) = _EFPLStat;

  factory EFPLStats.initial() => const EFPLStats(
        highestPoint: 0,
        averagePoint: 0,
        mostSelectedPlayer: "",
        mostTransferredOutPlayer: "",
        transfersMadeCount: 0,
        mostCaptainedPlayer: "",
        mostViceCaptainedPlayer: "",
        benchBoostCount: 0,
        freeHitCount: 0,
        wildCardCount: 0,
        tripleCaptainCount: 0,
        maxActiveCount: 0,
        gameWeekId: 0,
        dreamTeam: {},
      );
}
